### PR TITLE
New nanny: check Shares' affinity/anti affinity relationships

### PR DESCRIPTION
**Pull request by chuan137**

### CODE REVIEW

File: scripts/manila-check-affinity.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                for s in share_hosts:
+                    LOG.error("Share %s on host %s", s['share_id'], s['host'])
+
+        self.antiAffinityGauge.export(violated_rules)
+
+        LOG.info("Checking Affinity Rules")
+
+        self._update_affinity_groups()
+
+        violated_rules = []
```
Comment (Carthaca) [inline scripts/manila-check-affinity.py:61]: I would prefer a new variable name instead of recycling

File: scripts/manila-check-affinity.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+        self._update_anti_affinity_groups()
+
+        violated_rules = []
+        for grp in self.anti_affinity_groups:
+            check, share_hosts = self._check_group(grp, attracting=False)
+            if not check:
+                LOG.error("Anti-Affinity rule violated for Shares")
+                for s in share_hosts:
+                    LOG.error("Share %s on host %s", s['share_id'], s['host'])
```
Comment (Carthaca) [inline scripts/manila-check-affinity.py:53]: violated_rules need to be filled in here, otherwise the antiAffinityGauge is always empty

**Review by Carthaca — CHANGES_REQUESTED** [review]
Thanks for adding these checks!

I think this would also check soft-deleted shares or am I missing something?
The db queries could get a `deleted='False'` or `deleted=0`  condition respectively

**Comment by chuan137** [comment]
> I think this would also check soft-deleted shares or am I missing something? The db queries could get a `deleted='False'` or `deleted=0` condition respectively

thanks, overlooked the conditions. queries updated

**Review by Carthaca — APPROVED** [review]
Thanks

**Pull request merged**